include/Realisierung.tex

```diff
@@ -189,21 +190,28 @@ \subsubsection{Tag Reader}
+\begin{figure}[htb]
+	\centering
+	\includegraphics[keepaspectratio,width=.8\linewidth]{AblaufdiagrammTagErkennung}
+	\caption{Ablaufdiagramm zur Tag-Erkennung mit Darstellung, welcher Ablauf in welcher Komponenten erfolgt}
+	\label{fig:AblaufdiagrammTagErkennung}
+\end{figure}
+
 \clearpage
 \subsection{Umsetzung Programmierung}
 \subsubsection{Clean Code}
-Da gut lesbarer Code verständlicher und zeitnaher die Aufgaben des Codes widerspiegelt
-als ein Kommentar, wurde grösstenteils auf JavaDoc und weitere Kommentare verzichtet.
-Dies hatte zur Folge, dass bei der Erstellung einer Methode notwendigerweise ein selbst
-sprechender Name für diese sowie deren Parameter gewählt werden musste.
+Gut lesbarer Code ist verständlicher als veralteter, nicht akkurater Kommentar. Diesem Zustand haben wir entgegengesetzt, indem wir auf JavaDoc, sowie nicht notwendige Kommentare verzichtet haben. Dieser Entscheid hatte zur Folge, dass eine sprechende Namensgebung für alle zu benennende Elemente im Code verwendet wurden. Weiter wurden starke Verschachtelungsebene in einzelne Funktionen unterteilt, sowie Klassen eher klein gehalten (\cite{martin2009clean}).
 
-\subsubsection{Git Flow}
-Um möglichst effizient und unabhängig voneinander zu programmieren, sowie zu dokumentieren, wurde mit Git Feature Branch Workflow gearbeitet. Dies bedeutet, dass für jedes Feature einen Branch erstellt wurde (\cite{gitFeatureBranchWorkflow}). Weiter wurde eine Regelung eingeführt, wobei ein Branch nicht vom Ersteller in den Master Branch zusammengeführt werden darf. Um diese regel zu erzwingen, wurde der Master Branch auf github für alle Projekte entsprechend geschützt. Durch diesen schritt konnten wir die Sicherheit erhöhen, da eine zweite Person ein Review der Dokumentation respektive des Codes macht.
-Auf der Abbildung \ref{fig:FeatureBranchWorkflow} ist ein Auszug aus der Baumstruktur unserer Dokumentation dargestellt. Dabei ist zu erkennen, dass zu Sprint-beginn möglichst keine offenen Zweige (engl. Branches) mehr vorhanden sind und gegen Sprintende alle Zweige in den Master (in Abbildung \ref{fig:FeatureBranchWorkflow} schwarz dargestellt) über einen Pull Request zusammengeführt werden.
+\subsubsection{Feature Branch Workflow}
+Um möglichst effizient und unabhängig voneinander zu programmieren, sowie zu dokumentieren, wurde mit dem Feature Branch Workflow gearbeitet. Dies bedeutet, dass für jedes Feature einen Branch erstellt wurde. Weiter wurde eine Regelung eingeführt, wobei ein Branch nicht vom Ersteller in den Master Branch zusammengeführt werden darf. Um diese Regel zu erzwingen, wurde der Master Branch auf GitHub für alle Projekte entsprechend geschützt. Durch diesen Schritt, konnten wir die Sicherheit erhöhen, da eine zweite Person ein Review der Dokumentation respektive des Codes durchführen muss.
```

Comment:
Weiter wurde eine Regelung eingeführt, wobei ein Branch nicht vom Ersteller ~in den~ __mit dem__ Master Branch zusammengeführt werden darf.

oder

Weiter wurde eine Regelung eingeführt, wobei ein Branch nicht vom Ersteller in den Master Branch ~zusammengeführt~ _eingepflegt / eingefügt_ werden darf.